src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -76,23 +78,18 @@ public static void generate(String... args) throws IOException, CheckstyleExcept
     private static void dumpMetadata(Checker checker, String path) throws CheckstyleException,
             IOException {
         final List<File> validFiles = new ArrayList<>();
-        if (path.endsWith(".java")) {
-            validFiles.add(new File(path));
```

Comment:
Removed testing single files which I assume was only used by UTs. Most of our needs scan folders.